src/ol_infrastructure/applications/bootcamps/__main__.py

```diff
@@ -0,0 +1,189 @@
+from ol_infrastructure.lib.vault import setup_vault_provider
+
+setup_vault_provider()
+bootcamps_config = Config("bootcamps")
+stack_info = parse_stack()
+network_stack = StackReference(f"infrastructure.aws.network.{stack_info.name}")
+apps_vpc = network_stack.require_output("applications_vpc")
+data_vpc = network_stack.require_output("data_vpc")
+operations_vpc = network_stack.require_output("operations_vpc")
+aws_config = AWSBase(
+    tags={
+        "OU": "bootcamps",
+        "Environment": f"applications_{stack_info.env_suffix}",
+        "Application": "bootcamps",
+    }
+)
+
+# Create S3 buckets
+
+# Bucket used to store file uploads from bootcamps app.
+bootcamps_storage_bucket_name = f"ol-bootcamps-app-{stack_info.env_suffix}"
+bootcamps_storage_bucket = s3.Bucket(
+    f"ol-bootcamps-app-{stack_info.env_suffix}",
+    bucket=bootcamps_storage_bucket_name,
+    versioning=s3.BucketVersioningArgs(
+        enabled=True,
+    ),
+    policy=json.dumps(
+        {
+            "Version": "2008-10-17",
+            "Statement": [
+                {
+                    "Effect": "Allow",
+                    "Principal": {"AWS": "*"},
+                    "Action": "s3:GetObject",
+                    "Resource": f"arn:aws:s3:::{bootcamps_storage_bucket_name}/courses/*",  # noqa: E501
+                }
+            ],
+        }
+    ),
+    tags=aws_config.tags,
+)
+
+
+bootcamps_iam_policy = iam.Policy(
+    f"bootcamps-{stack_info.env_suffix}-policy",
+    description="AWS access controls for the bootcamps application in the "
+    f"{stack_info.name} environment",
+    path=f"/ol-applications/bootcamps/{stack_info.env_suffix}/",
+    name_prefix="aws-permissions-",
+    policy=lint_iam_policy(
+        {
+            "Version": "2012-10-17",
+            "Statement": [
+                {
+                    "Effect": "Allow",
+                    "Action": "s3:ListAllMyBuckets",
+                    "Resource": "*",
+                },
+                {
+                    "Effect": "Allow",
+                    "Action": [
+                        "s3:*MultiPartUpload*",
+                        "s3:ListBucket*",
+                        "s3:PutObject*",
+                        "s3:GetObject*",
+                        "s3:DeleteObject*",
+                    ],
+                    "Resource": [
+                        f"arn:aws:s3:::{bootcamps_storage_bucket_name}",
+                        f"arn:aws:s3:::{bootcamps_storage_bucket_name}/*",
+                    ],
+                },
+            ],
+        },
+        stringify=True,
+        parliament_config={
+            "PERMISSIONS_MANAGEMENT_ACTIONS": {
+                "ignore_locations": [{"actions": ["s3:putobjectacl"]}]
+            },
+            "RESOURCE_EFFECTIVELY_STAR": {},
+        },
+    ),
+)
+
+bootcamps_vault_backend_role = vault.aws.SecretBackendRole(
+    f"bootcamps-app-{stack_info.env_suffix}",
+    name=f"bootcamps-app-{stack_info.env_suffix}",
```

Comment:
The role doesn't need to have the env information in it because of the fact that we have distinct Vault clusters per env-stage